packages/slate-editor/src/components/ImageWithLoadingPlaceholderV2.tsx

```diff
@@ -65,11 +74,21 @@ const ImageWithLoadingPlaceholderV2 = forwardRef<HTMLElement, Props>(
         },
         ref,
     ) => {
+        const onIsLoadingChangeRef = useLatest(onIsLoadingChange);
+        const [isLoadingDebounced, setIsLoadingDebounced] = useState(false);
         const { loading, progress, url } = useImage(src);
 
         useEffect(() => {
-            onIsLoadingChange(loading);
-        }, [loading, onIsLoadingChange]);
+            setIsLoadingDebounced(loading);
+        }, [loading]);
+
+        useDebounce(
+            () => {
+                onIsLoadingChangeRef.current(isLoadingDebounced);
+            },
+            IS_LOADING_CHANGE_DEBOUNCE,
+            [isLoadingDebounced],
+        );
```

Comment:
btw. my favorite library for debounced values: https://github.com/xnimorz/use-debounce, makes code cleaner